Modules/_winapi.c

```diff
@@ -1920,36 +1910,24 @@ static PyMethodDef winapi_functions[] = {
     {NULL, NULL}
 };
 
-static struct PyModuleDef winapi_module = {
-    PyModuleDef_HEAD_INIT,
-    "_winapi",
-    NULL,
-    -1,
-    winapi_functions,
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
 #define WINAPI_CONSTANT(fmt, con) \
     PyDict_SetItemString(d, #con, Py_BuildValue(fmt, con))
 
-PyMODINIT_FUNC
-PyInit__winapi(void)
+static int winapi_exec(PyObject *m)
 {
     PyObject *d;
-    PyObject *m;
+    WinApiState *st;
 
-    if (PyType_Ready(&OverlappedType) < 0)
-        return NULL;
+    st = winapi_get_state(m);
+
+    st->overlapped_type = (PyTypeObject *)PyType_FromModuleAndSpec(m, &winapi_overlapped_type_spec, NULL);
+    if (st->overlapped_type == NULL) {
+        return -1;
+    }
 
-    m = PyModule_Create(&winapi_module);
-    if (m == NULL)
-        return NULL;
     d = PyModule_GetDict(m);
 
-    PyDict_SetItemString(d, "Overlapped", (PyObject *) &OverlappedType);
+    PyDict_SetItemString(d, "Overlapped", (PyObject *) st->overlapped_type);
```

Comment:
You can use PyModule_AddType() here. And please add error checking.

Reply:
@vstinner I think you mean PyModule_AddObject here

Reply:
I mean PyModule_AddType().